lib/colorado_lottery.rb

```diff
@@ -0,0 +1,52 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants= Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def interested_and_18?(contestant, game)
+    interested = contestant.game_interests.any? { |interested| interested == game.name }
+    interested && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    in_state_or_national = (!contestant.out_of_state? || game.national_drawing)
+    interested_and_18?(contestant, game) && in_state_or_national
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
```

Comment:
you don't need to check for this-- the interaction pattern says we will only call this method for contestants that can be successfully registered. but good thought.